src/CliCore.ts

```diff
@@ -127,7 +127,7 @@ export default class CliCore {
         } else { // if we are re-initializing, we need to reset the tagSet and options, if provided
             this.templateProcessor.tagSet = new Set();
             this.templateProcessor.options = options;
-            if (contextData) {
+            if (contextData && Object.keys(contextData).length > 0) {
```

Comment:
no biggie, but by the code, contextData appears it would never be undefined since it defaults to {} when no contextFilePath